flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -160,4 +304,16 @@ public boolean isDistributedFS() {
     public FileSystemKind getKind() {
         return delegateFS.getKind();
     }
+
+    private Tuple2<Boolean, Path> tryBuildLocalPath(Path path) {
+        String remotePathStr = path.toString();
+        if (localFileFilter.apply(path.getName()) && remotePathStr.startsWith(remoteBase)) {
```

Comment:
`path` might be a directory. I'd suggest not considering the `localFileFilter` here

Reply:
Directories won't end with `.sst`, `localFileFilter.apply(path.getName())` is always true for directories.